main.js

```diff
@@ -79,6 +80,17 @@ app.on('ready', function () {
   ipcMain.on(constants.IPC_START_DOWNLOAD, (event, items, access_token) => {
     downloadManager(items, access_token);
   });
+
+  ipcMain.on(constants.IPC_VERIFY_BEFORE_DOWNLOAD, (event, destination, totalFilesSize) => {
```

Comment:
I'm curious why you're using IPC to implement this API instead of calling `handleDiskSpace()` directly.

Reply:
Since Electron uses Chromium for displaying web pages, Chromium's multi-process architecture is also used. Each web page in Electron runs in its own process, which is called the `renderer process`. 
So, in normal browsers, web pages usually run in a sandboxed environment and are not allowed to access to native resources. Electron users, however, have the power to use `Node.js` APIs in web pages allowing lower level operating system interactions.
In this case `handeDiskSpace()` uses a library to check free disk space that runs commands for specific operative system, being necessary to run this method on a lower level, which means that we need to go through IPC implementation. 

Here's some reference info:
https://electronjs.org/docs/tutorial/quick-start  -> the first part of this article explains diferences between renderer processes and main process.

https://www.npmjs.com/package/diskspace -> this is node library that we are using to check disk space, you can find a link to github repo if you feel like to check the code, its not very extensive, but here you can see why we need to use lower operative system level.

Reply:
Thanks for the explanation, that makes sense.